### DISCUSSION TITLE
Erfahrungsbericht, wie ich Open3E mit iobroker auf einem Raspberry Pi zum Laufen brachte

### DISCUSSION THREAD

**Discussion by Hotziplotzi** [open3e/open3e#5] [Show and tell]
Hier im folgenden meine kleine Doku zum erfolgreichen Einsatz von Open3E.
Ermöglicht durch die tolle, freundliche Hilfe der Menschen hier:
https://www.viessmann-community.com/t5/Konnektivitaet/CAN-Bus-Home-Automation-E3-Generation-lokal-und-kostenlos/td-p/356066/page/6

![CAN 1](https://github.com/abnoname/open3e/assets/146114965/344b9c66-1d2b-4537-867d-80111db6bae8)
![CAN 2](https://github.com/abnoname/open3e/assets/146114965/0d12d4c5-620d-4eeb-8c5b-69e14ef7f8a4)
![CAN 3](https://github.com/abnoname/open3e/assets/146114965/46e1fab4-a9b7-48e8-9706-5787268ec371)
![CAN 4](https://github.com/abnoname/open3e/assets/146114965/1c4ec979-53c7-46cd-9348-703f099e55d0)
![CAN 5](https://github.com/abnoname/open3e/assets/146114965/cd12bb5f-ba44-4ec5-9557-6a2c6bf012d2)
![CAN 6](https://github.com/abnoname/open3e/assets/146114965/3b73f524-1d90-452e-8c74-751e7b1fe485)

![CAN 1](https://github.com/user-attachments/assets/f5aba959-21f1-4e69-8398-91a23a243b18)
![CAN 2](https://github.com/user-attachments/assets/4e6641b3-f32b-49a5-aa4c-c9b428565a4a)
![CAN 3](https://github.com/user-attachments/assets/77f9f959-1f1d-4168-b6c5-1befbc39ef12)
![CAN 4](https://github.com/user-attachments/assets/7c79bf68-f710-43e4-adfb-df32fcf4f40b)
![CAN 5](https://github.com/user-attachments/assets/e3726385-82b6-4b6b-aa29-4f15d4f59c77)
![CAN 6](https://github.com/user-attachments/assets/05cf2d1c-604c-4fe1-b1a2-6ffbcc388f5f)

**Comment by Hotziplotzi** [comment]
Sorry, dieser Thread wurde von Github erst mit über 1 Woche verspätung sichtbar geschaltet. Da er jetzt "doppelt" ist, kann er weg.

**Reply by philippoo66** [reply]
nein, es kann wenn überhaupt der andre weg :-) aber so viel Platz wird github noch für uns übrig haben... ;-)

**Comment by Thyler21** [comment]
@Hotziplotzi Super Anleitung - ich hab´s zumindest soweit am laufen, dass ich die "did´s" bekomme - leider stecke ich noch an der mqtt weiterleitung fest - da hängt das script im Terminal bei 
`loxberry@loxberry:~/open3e $ python3 Open3Eclient.py -c can0 -dev vx3 -r 1836,1824,1834 -m 127.0.0.1:1883:open3e -t 10   
Read dids and publish to mqtt...  `
bei Read dids an publish to mqtt steht es dann - bekomm auch in den Boker des Loxberry kein topic der open3e

ich hab es auch mit folgenden Probiert:
`loxberry@loxberry:~/open3e $ python3 Open3Eclient.py -c can0 -dev vx3 -r 1836,1824,1834 -m 127.0.0.1:1883:open3e -t 10 -muser loxberry -mpass MQTTPASSWORT  
Read dids and publish to mqtt...  `
sowie:
`loxberry@loxberry:~/open3e $ python3 Open3Eclient.py -c can0 -dev vx3 -r 1836,1824,1834 -m 127.0.0.1:1883:open3e -t 10 -muser [loxberry] -mpass [MQTTPASSWORT]  
Read dids and publish to mqtt...  `
hab ebenfalls die IP des Loxberry verwendet - da aber dieser auf dem selben RaspPi läuft wurde oben die localhost genommen.

Irgendjemand eine Ahnung?

**Reply by philippoo66** [reply]
Moin @Thyler21 !

Sorry, in den neueren Versionen ist MQTT User und Password zusammengfasst zu `-muser USER:PASSWRD` (weil eins ohne das andre keinen Sinn macht und ich die args Flut etwas reduzieren wollte...)

**Reply by Thyler21** [reply]
Servus @philippoo66 !
Danke für die Info - jedoch ist es hier auch das selbe...
ev. noch eine Idee?
Die Abfrage der did´s funktioniert einwandfrei

hat jemand erfahrung mit dem Loxberry - kann es ev. daran liegen?

**Reply by MyHomeMyData** [reply]
@Thyler21 Funktioniert die Anmeldung am MQTT Broker mit den gleichen Daten mit einem anderen Client? Z.B.
```
mosquitto_pub -h localhost -u Username -P Pw -t open3e -m "Hello world"
```

**Reply by MyHomeMyData** [reply]
@Thyler21 Wir hatten ein sporadisches Problem beim Aufsetzen des MQTT Clients. Bin unsicher, ob das etwas mit Deinem Problem zu tun hat. Würdest Du bitte die neueste Version im Branch develop probieren?

**Reply by Thyler21** [reply]
Entschuldigungn für die späte Antwort:

mit der aktuellen Version und folgender Abfrage funktioniert es 
`python3 Open3Eclient.py -c can0 -dev vx3 -r 680,1801,1802,1836 -t 60 -m localhost:1883:open3e -mfstr {didNumber}_{didName} -muser USER:PASSWORD
`
hatte zwischendurch ein anmeldeproblem beim Broker - das lag jedoch an den Subscriptions - ich hab mal mittels /# alle aufgemacht und et voilà - ich hab die Daten auf meinem Loxberry

**Comment by jscheidtmann** [comment]
Hi, die Bilder oder was auch eingebunden war, liefern bei mir 404 Fehler. Gibt's die Beschreibung irgendwo auch komplett? Danke!

**Reply by philippoo66** [reply]
moin @jscheidtmann und frohes neues!

Das Projekt wurde ja verschoben und ich weiss auch nicht, wordauf die Links zeigten. Aber ich meine, wir haben Hotzis Beitrag in's Wiki übernommen. https://github.com/open3e/open3e/wiki/020-Inbetriebnahme-CAN-Adapter-am-Raspberry

**Reply by philippoo66** [reply]
ps. @jscheidtmann doch nicht ganz, aber ich glaube ich habe die Dateien wiedergefunden (und hab sie erneut eingefügt)

siehe auch https://github.com/open3e/open3e/wiki/095-ioBroker-Adapter

**Reply by jscheidtmann** [reply]
Danke! :-)
